test-cases/tests/alerts/c18-validate-api-usage-alerts.md

```diff
@@ -0,0 +1,38 @@
+---
+products:
+  - name: rhoam
+    environments:
+      - osd-fresh-install
+estimate: 1h
+tags:
+  - per-release
+---
+
+# C18 - Validate API usage alerts
+
+## Prerequisites
+
+1. Login via `oc` as a user with **cluster-admin** role (kubeadmin):
+
+```
+oc login --token=<TOKEN> --server=https://api.<CLUSTER_NAME>.s1.devshift.org:6443
+```
+
+2. RHOAM is installed
+
+## Steps
+
+1. Validate that rate-limiting alerts ConfigMap has been created
+
+```
+oc get cm rate-limit-alerts -n redhat-managed-api-operator
+```
+
+2. Verify that level1, level2 and level3 rate-limiting alerts are present
+
+```
+oc get prometheusrules -n redhat-managed-api-marin3r
+```
+
+3. (TODO: modify the alerts configmap - configmap name?)
```

Comment:
Assuming that this test case is performed on a cluster where RHOAM has not been installed via the add-on, we should be able to modify the `sku-limits-managed-api-service` configmap in the RHOAM operator namespace to something like:
```
data:
  rate_limit: |-
    {
      "RHOAM SERVICE SKU": {
        "unit": "minute",
        "requests_per_unit": 20
      }
    }
```
Once the rate limit configs are updated, we should then create a new 3Scale application by logging into 3scale-admin route (login creds can be retrieved from the system-seed secret) and follow the getting started wizard. This allows us to easily hit the rate limit percentages.

Useful command for running X number of requests per second:
```
for i in {1..20}; do curl -i https://api-3scale-apicast-staging.apps.<cluster-id>.devshift.org//?user_key=XXXX; done
```
@sergioifg94 might should have detailed steps on how to trigger each alert

Reply:
It seems that the steps for creating a new 3Scale application are listed in `h22-validate-that-rate-limit-service-is-working-as-expected.md` so we should be able to copy/paste them

Reply:
For now I've only managed to trigger the level 3 one, but I'm gonna try to play with the different configurations to get the other ones

Reply:
Thanks for the additional info guys.

Regarding checking the alerts - maybe this script that resiliency team is using for checking the alerts during performance testing could help us to easily detect the alerts?
https://github.com/integr8ly/integreatly-operator/blob/master/scripts/alerts-during-perf-testing.sh